src/sagemaker/estimator.py

```diff
@@ -614,16 +614,24 @@ def __init__(
         self.output_kms_key = resolve_value_from_config(
             output_kms_key, TRAINING_JOB_KMS_KEY_ID_PATH, sagemaker_session=self.sagemaker_session
         )
+        use_volume_kms_config: bool = False
         if instance_type is None or isinstance(instance_type, str):
             instance_type_for_volume_kms = instance_type
         elif isinstance(instance_type, ParameterString):
             instance_type_for_volume_kms = instance_type.default_value
```

Comment:
Can we remove this condition of `elif isinstance(instance_type, ParameterString):` since ParameterString is a type of PipelineVariable. They are all parsed only during pipeline execution time. 
Thus, if user change the ParameterString value when invoking `pipeline.start`, this condition here is no longer valid